src/util/strencodings.h

```diff
@@ -94,8 +94,11 @@ void SplitHostPort(std::string in, uint16_t& portOut, std::string& hostOut);
 // which provide parse error feedback.
 //
 // The goal of LocaleIndependentAtoi is to replicate the exact defined behaviour
-// of atoi and atoi64 as they behave under the "C" locale.
-template <typename T>
+// of atoi as it behaves under the "C" locale.
+//
+// Prevent use of LocaleIndependentAtoi for int64_t, since the function below
```

Comment:
In commit "Restore atoi64 compatibility with old versions of Bitcoin Core" (ac1a5b113ccdace4d7f04e2e8c4089e397b5c0b4)

I don't understand why this function exists at all. What use could there be for a parsing function that returns 0 in the case of overflow? It just seems like a footgun. Returning the biggest possible value when a bigger value is given, and the smallest possible value when a smaller value is given seems like a more useful and less dangerous way to parse values.

I guess it precedes this PR, but the naming also seems strange. Why call this one util function locale independent, when basically every util function we have should be locale independent? If we had use for a parsing function that did vary by locale it would make more sense to draw attention and mention locales in **that** function's name, not this function's name.

Reply:
I like that the name is long and ugly. It makes developers less likely to pick it, when there are better named and safer alternatives available that come with error reporting instead of implicitly mapping errors to random (in-range) return values.

Reply:
re: https://github.com/bitcoin/bitcoin/pull/23841#discussion_r781924579

> I like that the name is long and ugly. It makes developers less likely to pick it, when there are better named and safer alternatives available that come with error reporting instead of implicitly mapping errors to random (in-range) return values.

It's just seems a weird way to make the function name long by adding a random fact about locale independence. You could call it `FrogsAreGreenAtoi` and that would be better name than `LocaleIndependentAtoi` because it wouldn't falsely suggest that other functions depend on the current locale and this function doesn't. I think an ideal name would be something more like `ParseIntUnchecked` or `AtoiIgnoreErrors`

Reply:
> FrogsAreGreenAtoi

Ryan, not all frogs are green. I'd suggest to use the name `MyFrogIsGreenAtoi` or `ParseSaturatedInt`.

Reply:
+1 to `MyFrogIsGreenAtoi`